docs/adr/0002-serialize-complex-datatypes.md

```diff
@@ -0,0 +1,101 @@
+# The MVP will pass complex datatypes over the FFI using simple, explicit serialization
+
+* Status: accepted
+* Deciders: rfkelly, YOUR NAME HERE
+* Date: 2020-07-01 (or thereabouts)
```

Comment:
This ADR is mostly in response to recent discussion with Thom about the perceived overheads of our current approach; here's the [rendered view](https://github.com/mozilla/uniffi-rs/blob/docs-and-adrs/docs/adr/0002-serialize-complex-datatypes.md) for your convenience.

I think it's important to be explicit that we're defaulting to simple-safe-and-slow where possible for the first version, but that it doesn't have to stay that way forever. This is also the one place where we're diverging from what has become best-practices in app-services code, so I figured that was worth an explanation.